### DISCUSSION TITLE
Make text wrapping API puplic

### DISCUSSION THREAD

**Discussion by h0lg** [dotnet/command-line-api#2286] [Ideas]
When building CLI apps, wrapping large text to fit the `Console.WindowWidth` is a common requirement. It would be nice if an API like `HelpBuilder.WrapText(string text, int maxWidth)` would be publicly accessible for reuse.

https://github.com/dotnet/command-line-api/blob/436f97374ca2631a2e8eb9bfe99082a75da351da/src/System.CommandLine/Help/HelpBuilder.cs#L319

If there's another API to do that, please point me in the right direction, thanks!

**Comment by KalleOlaviNiemitalo** [comment]
It doesn't currently measure widths correctly (<https://github.com/dotnet/command-line-api/issues/1973>), and fixing that may require lookup tables that would then cause the package to grow more than is desired for some applications.

If a string wrapping API is made public, then I think it should be defined with a delegate, interface, or abstract class, so that

- System.CommandLine provides a simple implementation and uses that by default
- Applications can also call that implementation
- Applications can also provide a more complex implementation (supporting wide characters, emoji, CJK word boundaries without spaces, etc.) and configure System.CommandLine to use that

**Reply by h0lg** [reply]
Thanks for your input. I understand that before making a text wrapping API public, you suggest waiting for the API changes that the resolution of #1973 will probably have to make. Makes sense.
